core/testcontainers/core/network.py

```diff
@@ -0,0 +1,47 @@
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import uuid
+from contextlib import suppress
+from typing import Optional
+
+from testcontainers.core.docker_client import DockerClient
+
+
+class Network:
+    """
+    Network context manager for programmatically connecting containers.
+    """
+
+    def __init__(self, docker_client_kw: Optional[dict] = None, **kwargs) -> None:
+        self.name = str(uuid.uuid4())
+        self._docker = DockerClient(**(docker_client_kw or {}))
+        self._kwargs = kwargs
+
+    def connect(self, container_id: str, network_aliases: Optional[list] = None):
+        self._network.connect(container_id, aliases=network_aliases)
+
+    def remove(self) -> None:
+        self._network.remove()
+
+    def __enter__(self) -> "Network":
+        self._network = self._docker.client.networks.create(self.name, **self._kwargs)
+        self.id = self._network.id
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> None:
+        self.remove()
+
+    def __del__(self) -> None:
```

Comment:
We've had some bad experiences with external resource cleanup through the `__del__` method, so we have decided to move away from this. Having the context manager and the `remove` method provides sufficient means to cleanup the resource. `__del__` hook can be removed 👍